libsolidity/ast/Types.cpp

```diff
@@ -3132,7 +3131,10 @@ FunctionTypePointer FunctionType::interfaceFunctionType() const
 {
 	// Note that m_declaration might also be a state variable!
 	solAssert(m_declaration, "Declaration needed to determine interface function type.");
-	bool isLibraryFunction = kind() != Kind::Event && dynamic_cast<ContractDefinition const&>(*m_declaration->scope()).isLibrary();
+	bool isLibraryFunction = false;
+	if (kind() != Kind::Event)
+		if (auto const* contract = dynamic_cast<ContractDefinition const*>(m_declaration->scope()))
+			isLibraryFunction = contract->isLibrary();
```

Comment:
Can this be called by free functions?

Reply:
Calling `interfaceFunctionType` on a free function does not make sense, but I think it can happen inside the type system at some point. The bool `isLibraryFunction` would be false in this case. Do you think it is a problem?